nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConnection.java

```diff
@@ -244,4 +266,71 @@ private String certsToString(Certificate[] peerCertificates) {
+                connection.helidonSocket = PlainSocket.client(connection.socket, connection.channelId);
+            }
+        },
+        HTTPS {
+            @Override
+            protected void connect(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException {
+                SSLSocket sslSocket = connection.connectionKey.tls().createSocket("http/1.1");
+                connection.socket = sslSocket;
+                connection.socket.connect(remoteAddress, (int) connection.options.connectTimeout().toMillis());
+                sslSocket.startHandshake();
+                connection.helidonSocket = TlsSocket.client(sslSocket, connection.channelId);
+                if (LOGGER.isLoggable(TRACE)) {
+                    connection.debugTls(sslSocket);
+                }
+            }
+        },
+        PROXY_PLAIN {
+            @Override
+            protected void connect(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException {
+                connection.socket.connect(connection.connectionKey.proxy().address(),
+                        (int) connection.options.connectTimeout().toMillis());
+                int responseCode = connection.connectToProxy(remoteAddress);
+                if (responseCode != Status.OK_200.code()) {
+                    throw new IllegalStateException("Proxy sent wrong HTTP response code: " + responseCode);
+                }
+                connection.helidonSocket = PlainSocket.client(connection.socket, connection.channelId);
+            }
+        },
+        PROXY_HTTPS {
+            @Override
+            protected void connect(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException {
+                connection.socket.connect(connection.connectionKey.proxy().address(),
+                        (int) connection.options.connectTimeout().toMillis());
+                int responseCode = connection.connectToProxy(remoteAddress);
+                if (responseCode != Status.OK_200.code()) {
+                    throw new IllegalStateException("Proxy sent wrong HTTP response code: " + responseCode);
+                }
+                SSLSocket sslSocket = connection.connectionKey.tls().createSocket("http/1.1", connection.socket, remoteAddress);
+                connection.socket = sslSocket;
+                sslSocket.startHandshake();
+                connection.helidonSocket = TlsSocket.client(sslSocket, connection.channelId);
+                if (LOGGER.isLoggable(TRACE)) {
+                    connection.debugTls(sslSocket);
+                }
+            }
+        };
+
+        protected abstract void connect(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException;
+
+        static void configure(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException {
+            EstablishConnection connector = null;
```

Comment:
not needed null assignment.

Reply:
I don't know from which JDK this is allowed, but good to know. Done

Reply:
You only need to assign `null` (or any other initialization) if the variable is not assigned in any of the possible flows. This is since ever.